### PR TITLE
[CAMEL-20139]: fix for incorrect correlation key in first aggregated message

### PR DESCRIPTION
**Pull request by dinurp**
# Description

fix for
[[CAMEL-20139](https://issues.apache.org/jira/browse/CAMEL-20139)]: aggregate EIP: wrong correlation key set for the first aggregate exchange

When completing aggregation, original message of the completing message was passed as original messages for all batches. Now the completing message is passed as the original message only for its batch; for others, the aggregate message in the repository is passed as the original message.

# Target

- [X] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [X] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL-20139)

# Apache Camel coding standards and style

- [X] I checked that each commit in the pull request has a meaningful subject line and body.

- [X] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AggregateCompletionByBatchConsumerTest extends ContextTestSupport {
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCorrelationKey() throws Exception {
+        // START SNIPPET: e2
```
Comment (oscerd) [inline core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java:36]: Please remove SNIPPET

File: core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        out = result.getExchanges().get(0);
+        grouped = out.getIn().getBody(List.class);
+
+        assertEquals(2, grouped.size());
+
+        assertEquals("batch-1", grouped.get(0).getBody(String.class));
+        assertEquals("batch-1", grouped.get(1).getBody(String.class));
+        assertEquals("batch-1", out.getProperty(Exchange.AGGREGATED_CORRELATION_KEY));
+        // END SNIPPET: e2
```
Comment (oscerd) [inline core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java:95]: Same

File: core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Override
+    protected RouteBuilder createRouteBuilder() throws Exception {
+        return new RouteBuilder() {
+            public void configure() throws Exception {
+                // START SNIPPET: e1
```
Comment (oscerd) [inline core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java:102]: Same

File: core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                // the response to the mock
+                from("direct:start")
+                        // aggregate all using body and group the
+                        // exchanges so we get one single exchange containing all
+                        .aggregate(body(), new GroupedMessageAggregationStrategy())
+                        // we are simulating a batch consumer
+                        .completionFromBatchConsumer()
+                        .eagerCheckCompletion()
+                        .to("mock:result");
+                // END SNIPPET: e1
```
Comment (oscerd) [inline core/camel-core/src/test/java/org/apache/camel/processor/aggregator/AggregateCompletionByBatchConsumerTest.java:113]: Same

**Review by oscerd — CHANGES_REQUESTED** [review]
Just minor findings.

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Pull request merged**